common/scala/src/main/scala/whisk/core/entity/Exec.scala

```diff
@@ -43,40 +47,49 @@ import whisk.core.entity.ArgNormalizer.trim
  *         jar   : a base64-encoded JAR file when kind is "java",
  *         name  : a fully-qualified class name when kind is "java" }
  */
-sealed abstract class Exec(val kind: String) {
+sealed abstract class Exec(val kind: String) extends ByteSizeable {
     def image: String
 
     override def toString = Exec.serdes.write(this).compactPrint
 }
 
 protected[core] case class NodeJSExec(code: String, init: Option[String]) extends Exec(Exec.NODEJS) {
     val image = "whisk/nodejsaction"
+    def size = (code sizeInBytes) + init.map(_.sizeInBytes).getOrElse(0 B)
 }
 
 protected[core] case class NodeJS6Exec(code: String, init: Option[String]) extends Exec(Exec.NODEJS6) {
     val image = "whisk/nodejs6action"
+    def size = (code sizeInBytes) + init.map(_.sizeInBytes).getOrElse(0 B)
 }
 
 protected[core] case class PythonExec(code: String) extends Exec(Exec.PYTHON) {
     val image = "whisk/pythonaction"
+    def size = code sizeInBytes
 }
 
 protected[core] case class SwiftExec(code: String) extends Exec(Exec.SWIFT) {
     val image = "whisk/swiftaction"
+    def size = code sizeInBytes
 }
 
 protected[core] case class Swift3Exec(code: String) extends Exec(Exec.SWIFT3) {
     val image = "whisk/swift3action"
+    def size = code sizeInBytes
 }
 
 protected[core] case class JavaExec(jar: String, main: String) extends Exec(Exec.JAVA) {
     val image = "whisk/javaaction"
+    def size = (jar sizeInBytes) + (main sizeInBytes)
 }
 
-protected[core] case class BlackBoxExec(image: String) extends Exec(Exec.BLACKBOX)
+protected[core] case class BlackBoxExec(image: String) extends Exec(Exec.BLACKBOX) {
+    def size = image sizeInBytes
+}
 
 protected[core] case class SequenceExec(code: String, components: Vector[String]) extends Exec(Exec.SEQUENCE) {
     val image = "whisk/nodejsaction"
+    def size = (code sizeInBytes) + components.map(_ sizeInBytes).reduce(_ + _)
```

Comment:
the code part of a sequence is "free" - it shouldn't count against the action. (will make the change as part of the merge.)